### PR TITLE
INT-272 : eliogbility V2 - no badge

### PR DESCRIPTION
**Pull request by Francois-Gomis**

### CODE REVIEW

File: src/app/code/community/Alma/Installments/Helper/Eligibility.php [hunk 1]
```diff
@@ -27,143 +27,144 @@
...
+        $eligibleFeePlans = $this->selectEligibleFeePlans($feePlansEligibilities);
+        $this->saveEligibleFeePlans($eligibleFeePlans);
+        return $eligibleFeePlans;
+    }
 
-        if (!$this->alma) {
-            $this->eligible = false;
-            return false;
+    /**
+     * @return array|void
```
Comment (syjust-alma) [inline src/app/code/community/Alma/Installments/Helper/Eligibility.php]: this method does not return `void`

Comment (syjust-alma) [inline src/app/code/community/Alma/Installments/Helper/Eligibility.php]: `void` seem's not be a possible `@return`

Then you can give a type for the array returned like this `array|Eligibility[]` (and try to give a type everytime you return an array if you which kind of objects are present in array 🙏 )

File: src/app/code/community/Alma/Installments/Helper/Eligibility.php [hunk 2]
```diff
@@ -27,143 +27,144 @@
...
+        return $feePlansEligibilities;
+    }
 
-        $this->message = $eligibilityMessage;
-        $cartTotal = Alma_Installments_Helper_Functions::priceToCents((float)$quote->getGrandTotal());
+    /**
+     * @param $eligibleFeePlans
+     * @return void
+     */
+    private function saveEligibleFeePlans($eligibleFeePlans)
```
Comment (syjust-alma) [inline src/app/code/community/Alma/Installments/Helper/Eligibility.php]: I think you can merge this behavior directly in `getAlmaFeePlansEligibility()` method

File: src/app/code/community/Alma/Installments/Helper/Eligibility.php [hunk 3]
```diff
@@ -27,143 +27,144 @@
...
-                    $this->message .= ' ' . sprintf($this->__('(Maximum amount: %s)'), $price);
-                } else {
-                    $price = $this->getFormattedPrice(Alma_Installments_Helper_Functions::priceFromCents($minAmount));
-                    $this->message .= ' ' . sprintf($this->__('(Minimum amount: %s)'), $price);
-                }
+    /**
+     * @param $feePlansEligibilities
+     * @return array
+     */
+    private function selectEligibleFeePlans($feePlansEligibilities){
```
Comment (syjust-alma) [inline src/app/code/community/Alma/Installments/Helper/Eligibility.php]:
```suggestion
    private function filterEligibleFeePlans($feePlansEligibilities){
```

File: src/app/code/community/Alma/Installments/Helper/Eligibility.php [hunk 4]
```diff
@@ -27,143 +27,144 @@
...
+    }
 
-        if (!$this->alma) {
-            $this->eligible = false;
-            return false;
+    /**
+     * @return array|void
+     * @throws Mage_Core_Exception
+     */
+    private function getAlmaFeePlansEligibility()
```
Comment (syjust-alma) [inline src/app/code/community/Alma/Installments/Helper/Eligibility.php]: in my opinion, a `getter` should not make a complex action (just return a property)

maybe you can rename this method `loadEligibleFeePlans` ?

File: src/app/code/community/Alma/Installments/Helper/FeePlansHelper.php [hunk 5]
```diff
@@ -0,0 +1,317 @@
...
+        return [
+            self::MIN_DISPLAY_KEY,
+            self::MAX_DISPLAY_KEY
+        ];
+    }
+
+    /**
+     * @param $data
+     * @param $type
+     * @return mixed
```
Comment (syjust-alma) [inline src/app/code/community/Alma/Installments/Helper/FeePlansHelper.php]:
```suggestion
     * @return array
```

File: src/app/code/community/Alma/Installments/Helper/FeePlansHelper.php [hunk 6]
```diff
@@ -0,0 +1,317 @@
...
+     */
+    private function getFeePlansPriceKeysToConvertForSave(){
+        return [
+            self::MIN_DISPLAY_KEY,
+            self::MAX_DISPLAY_KEY
+        ];
+    }
+
+    /**
+     * @param $data
```
Comment (syjust-alma) [inline src/app/code/community/Alma/Installments/Helper/FeePlansHelper.php]: please fix with just one `@param` & add type

File: src/app/code/community/Alma/Installments/Helper/FeePlansHelper.php [hunk 7]
```diff
@@ -0,0 +1,317 @@
...
+        foreach ($configFeePlans as $feePlanKey=> $almaFeePlan) {
+            if(isset($formFeePlans[$feePlanKey])){
+            $configFeePlans[$feePlanKey] = $this->includeFormDataInFeePlan($almaFeePlan,$formFeePlans[$feePlanKey]);
+            }
+         }
+        return $configFeePlans;
+    }
+
+    /**
+     * @param $almaFeePlan
```
Comment (syjust-alma) [inline src/app/code/community/Alma/Installments/Helper/FeePlansHelper.php]: please give type for this 2 `@param`

File: src/app/code/community/Alma/Installments/Helper/FeePlansHelper.php [hunk 8]
```diff
@@ -0,0 +1,317 @@
...
+        foreach ($feePlans as $feePlan) {
+            if($feePlan['enable']){
+                $enabledFeePlans[$feePlan['id']] = $feePlan;
+            }
+        }
+        return $enabledFeePlans;
+    }
+
+    /**
+     * @param $configFeePlan
```
Comment (syjust-alma) [inline src/app/code/community/Alma/Installments/Helper/FeePlansHelper.php]: please give type for this 2 `@param`

File: src/app/code/community/Alma/Installments/Helper/FeePlansHelper.php [hunk 9]
```diff
@@ -0,0 +1,317 @@
...
+    {
+        foreach ($almaFeePlans as $index => $almaFeePlan) {
+            if ($almaFeePlan->getPlanKey() == self::ONE_INSTALLMENT_KEY){
+                unset($almaFeePlans[$index]);
+            }
+        }
+    }
+
+    /**
+     * @param $almaFormattedFeePlans
```
Comment (syjust-alma) [inline src/app/code/community/Alma/Installments/Helper/FeePlansHelper.php]: please give type for this `@param`

File: src/app/code/community/Alma/Installments/Helper/FeePlansHelper.php [hunk 10]
```diff
@@ -0,0 +1,317 @@
...
+            'custom_min_purchase_amount'=> $almaFeePlan->min_purchase_amount,
+            'custom_max_purchase_amount'=> $almaFeePlan->max_purchase_amount,
+            'max_purchase_amount'=>  $almaFeePlan->max_purchase_amount,
+            'deferred_trigger_limit_days'=> $almaFeePlan->deferred_trigger_limit_days,
+            'merchant_fee_variable'=> $almaFeePlan->merchant_fee_variable,
+        ];
+    }
+
+    /**
+     * @param $almaFeePlans (warning reference)
```
Comment (syjust-alma) [inline src/app/code/community/Alma/Installments/Helper/FeePlansHelper.php]: please give type for this `@param`

File: src/app/code/community/Alma/Installments/Helper/FeePlansHelper.php [hunk 11]
```diff
@@ -0,0 +1,317 @@
...
+    const ALMA_PNX_CONFIG_FEE_PLANS_PATH = 'payment/alma_installments/pnx_config';
+    const MIN_PURCHASE_AMOUNT_KEY = 'min_purchase_amount';
+    const MIN_DISPLAY_KEY = 'custom_min_purchase_amount';
+    const MAX_DISPLAY_KEY = 'custom_max_purchase_amount';
+    const MAX_PURCHASE_AMOUNT_KEY = 'max_purchase_amount';
+    const FEE_PLAN_ENABLE_KEY = 'enable';
+
+    private $almaClient;
+    /**
+     * @var
```
Comment (syjust-alma) [inline src/app/code/community/Alma/Installments/Helper/FeePlansHelper.php]: please give type for this `@var`

File: src/app/code/community/Alma/Installments/Helper/FeePlansHelper.php [hunk 12]
```diff
@@ -0,0 +1,317 @@
...
+    const ONE_INSTALLMENT_KEY = "general_1_0_0";
+    const ALMA_BASE_FEE_PLANS_PATH = 'payment/alma_installments/base_fee_plans';
+    const ALMA_PNX_CONFIG_FEE_PLANS_PATH = 'payment/alma_installments/pnx_config';
+    const MIN_PURCHASE_AMOUNT_KEY = 'min_purchase_amount';
+    const MIN_DISPLAY_KEY = 'custom_min_purchase_amount';
+    const MAX_DISPLAY_KEY = 'custom_max_purchase_amount';
+    const MAX_PURCHASE_AMOUNT_KEY = 'max_purchase_amount';
+    const FEE_PLAN_ENABLE_KEY = 'enable';
+
+    private $almaClient;
```
Comment (syjust-alma) [inline src/app/code/community/Alma/Installments/Helper/FeePlansHelper.php:38]: please give type for this `@var`

File: src/app/code/community/Alma/Installments/Helper/Eligibility.php [hunk 13]
```diff
@@ -172,16 +173,53 @@ private function getFormattedPrice($price)
...
             if (in_array($item->getRealProductType(), $excludedProductTypes)) {
+                $this->logger->error('A product is in excluded product type list',[]);
                 return false;
             }
         }
+        return true;
+    }
+
+    /**
+     * @param $quote
```
Comment (syjust-alma) [inline src/app/code/community/Alma/Installments/Helper/Eligibility.php:179]: please give type for this 2 `@param`

File: src/app/code/community/Alma/Installments/Helper/Eligibility.php [hunk 14]
```diff
@@ -27,143 +27,144 @@
...
-            if ($cartTotal < $minAmount || $cartTotal > $maxAmount) {
-                if ($cartTotal > $maxAmount) {
-                    $price = $this->getFormattedPrice(Alma_Installments_Helper_Functions::priceFromCents($maxAmount));
-                    $this->message .= ' ' . sprintf($this->__('(Maximum amount: %s)'), $price);
-                } else {
-                    $price = $this->getFormattedPrice(Alma_Installments_Helper_Functions::priceFromCents($minAmount));
-                    $this->message .= ' ' . sprintf($this->__('(Minimum amount: %s)'), $price);
-                }
+    /**
+     * @param $feePlansEligibilities
```
Comment (syjust-alma) [inline src/app/code/community/Alma/Installments/Helper/Eligibility.php]: please give type for this `@param`

File: src/app/code/community/Alma/Installments/Helper/Eligibility.php [hunk 15]
```diff
@@ -27,143 +27,144 @@
...
+            $this->logger->error('$e',[$e->getMessage()]);
+            return [];
         }
+        return $feePlansEligibilities;
+    }
 
-        $this->message = $eligibilityMessage;
-        $cartTotal = Alma_Installments_Helper_Functions::priceToCents((float)$quote->getGrandTotal());
+    /**
+     * @param $eligibleFeePlans
```
Comment (syjust-alma) [inline src/app/code/community/Alma/Installments/Helper/Eligibility.php]: please give type for this `@param`

File: src/app/code/community/Alma/Installments/Block/PaymentForm.php [hunk 16]
```diff
@@ -26,41 +26,104 @@
...
+    }
 
-		if (in_array(3, $availableInstallmentsCounts)) {
-			return 3;
-		} else {
-			$n = $availableInstallmentsCounts[count($availableInstallmentsCounts) - 1];
-			return $n;
-		}
+    /**
+     * @param $creditCost
```
Comment (syjust-alma) [inline src/app/code/community/Alma/Installments/Block/PaymentForm.php]: please give type for this `@param`

File: src/app/code/community/Alma/Installments/Block/PaymentForm.php [hunk 17]
```diff
@@ -26,41 +26,104 @@
...
+    public function getAnnualInterestRate($annualInterestRate)
+    {
+        return $annualInterestRate/100 .'%' ;
     }
 
-	public function defaultInstallmentsCount()
-	{
-		$availableInstallmentsCounts = $this->availableInstallmentsCounts();
+    /**
+     * @param $creditCost
```
Comment (syjust-alma) [inline src/app/code/community/Alma/Installments/Block/PaymentForm.php]: please give type for this `@param`

File: src/app/code/community/Alma/Installments/Block/PaymentForm.php [hunk 18]
```diff
@@ -26,41 +26,104 @@
...
+     * @param $cents
+     * @return mixed
+     */
+    public function convertCentToPrice($cents)
+    {
+        return Mage::helper('core')->currency($this->functionsHelper->priceFromCents($cents), true, false);
+    }
+
+    /**
+     * @param $fee
```
Comment (syjust-alma) [inline src/app/code/community/Alma/Installments/Block/PaymentForm.php]: please give type for this `@param`

File: src/app/code/community/Alma/Installments/Block/PaymentForm.php [hunk 19]
```diff
@@ -26,41 +26,104 @@
...
+     * @param $ts
+     * @return string
+     */
+    public function tsToLocaleDate($ts)
+    {
+        return Mage::app()->getLocale()->date($ts)->toString(Zend_Date::DATE_MEDIUM);
+    }
+
+    /**
+     * @param $cents
```
Comment (syjust-alma) [inline src/app/code/community/Alma/Installments/Block/PaymentForm.php]: please give type for this `@param`

File: src/app/code/community/Alma/Installments/Block/PaymentForm.php [hunk 20]
```diff
@@ -26,41 +26,104 @@
...
+            $eligibleFeePlans = $this->eligibilityHelper->getEligibleFeePlans();
+        } catch (Exception $e) {
+            $this->logger->error('Get Alma fee plans eligibility exception : ',[$e->getMessage()]);
+        }
+        $this->logger->info('$eligibleFeePlans',[$eligibleFeePlans]);
+        return $eligibleFeePlans;
+    }
+
+    /**
+     * @param $ts
```
Comment (syjust-alma) [inline src/app/code/community/Alma/Installments/Block/PaymentForm.php]: please give type for this `@param` and give a more eloquent parameter and method name (did you mean `timestamp` ?)

File: src/app/code/community/Alma/Installments/Helper/Eligibility.php [hunk 21]
```diff
@@ -27,143 +27,144 @@
...
-        $this->eligibilities = array();
+        $this->config = Mage::helper('alma/config');
+        $this->feePlansHelper = Mage::helper('alma/FeePlansHelper');
+        $this->currentEligibleFeePlans = array();
+        $this->eligibleFeePlansAreLoaded = false;
     }
 
     /**
-     * @return bool
+     * @throws Mage_Core_Exception
```
Comment (syjust-alma) [inline src/app/code/community/Alma/Installments/Helper/Eligibility.php:69]: add `@return`

File: src/app/code/community/Alma/Installments/Helper/FeePlansHelper.php [hunk 22]
```diff
@@ -0,0 +1,317 @@
...
+        $valueForLabel = $installments;
+
+        if ($deferred > 0 ){
+            $stringForLabel = 'Payment in %s days';
+            $valueForLabel = $deferred;
+        }
+        return sprintf($stringForLabel,$valueForLabel);
+    }
+
+    public function validateFeePlanMinAndMaxCustomAmount($feePlans)
```
Comment (syjust-alma) [inline src/app/code/community/Alma/Installments/Helper/FeePlansHelper.php:304]: please add a phpdoc block here

File: src/app/code/community/Alma/Installments/Model/System/Config/Backend/PnxConfig.php [hunk 23]
```diff
@@ -0,0 +1,90 @@
...
+    private $feePlansHelper;
+    public function __construct()
+    {
+        $this->feePlansHelper = Mage::helper('alma/FeePlansHelper');
+
+        parent::__construct();
+    }
+
+
+    public function _afterLoad()
```
Comment (syjust-alma) [inline src/app/code/community/Alma/Installments/Model/System/Config/Backend/PnxConfig.php]: can you add phpdoc with a description of what is done inside this inherited method please

you can also add `@override` if this method is not abstract in parent class

then, I think this method should be `protected` .... WDYT ?

File: src/app/code/community/Alma/Installments/Model/System/Config/Backend/PnxConfig.php [hunk 24]
```diff
@@ -0,0 +1,90 @@
...
+ * @copyright  Copyright (c) 2006-2020 Magento, Inc. (http://www.magento.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+/**
+ * Adminhtml catalog inventory "Minimum Qty Allowed in Shopping Cart" field
+ *
+ * @category   Mage
+ * @package    Mage_CatalogInventory
+ * @author     Magento Core Team <core@magentocommerce.com>
```
Comment (syjust-alma) [inline src/app/code/community/Alma/Installments/Model/System/Config/Backend/PnxConfig.php]: really ?

File: src/app/code/community/Alma/Installments/Model/System/Config/Backend/PnxConfig.php [hunk 25]
```diff
@@ -0,0 +1,90 @@
...
+ * to license@magento.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade Magento to newer
+ * versions in the future. If you wish to customize Magento for your
+ * needs please refer to http://www.magento.com for more information.
+ *
+ * @category    Mage
+ * @package     Mage_CatalogInventory
```
Comment (syjust-alma) [inline src/app/code/community/Alma/Installments/Model/System/Config/Backend/PnxConfig.php]: hum 🧐  ...

File: src/app/code/community/Alma/Installments/Model/System/Config/Backend/PnxConfig.php [hunk 26]
```diff
@@ -0,0 +1,90 @@
...
+            $feePlansWithPriceForDisplay = $this->feePlansHelper->convertFeePlansPricesForDisplay($mergedConfigAndFormFeePlans);
+
+            $this->setValue($feePlansWithPriceForDisplay);
+        }
+    }
+    /**
+     * Check object existence in incoming data and unset array element with '__empty' key
+     *
+     * @throws Mage_Core_Exception
+     * @return void
```
Comment (syjust-alma) [inline src/app/code/community/Alma/Installments/Model/System/Config/Backend/PnxConfig.php:67]: use `@override` here if the parent method is not abstract

File: src/app/code/community/Alma/Installments/Model/System/Config/Backend/PnxIsEnable.php [hunk 27]
```diff
@@ -0,0 +1,31 @@
+<?php
+
+
+
+class Alma_Installments_Model_System_Config_Backend_PnxIsEnable  extends Mage_Core_Block_Html_Select
+{
+    /**
+     * Render output
```
Comment (syjust-alma) [inline src/app/code/community/Alma/Installments/Model/System/Config/Backend/PnxIsEnable.php:29]: add type here please

File: src/app/code/community/Alma/Installments/Model/System/Config/Backend/PnxIsEnable.php [hunk 28]
```diff
@@ -0,0 +1,31 @@
...
+        foreach (Mage::getSingleton('adminhtml/system_config_source_yesno')->toOptionArray() as $option) {
+            $html.=$this->_optionToHtml($option);
+        }
+        $html.='</select>';
+
+        return  $html;
+    }
+
+    /**
+     * @param $value
```
Comment (syjust-alma) [inline src/app/code/community/Alma/Installments/Model/System/Config/Backend/PnxIsEnable.php]: add type here please

File: src/app/code/community/Alma/Installments/Model/System/Config/Backend/PnxStringRender.php [hunk 29]
```diff
@@ -0,0 +1,19 @@
+<?php
+
+
+
+class Alma_Installments_Model_System_Config_Backend_PnxStringRender  extends Mage_Core_Block_Abstract
```
Comment (syjust-alma) [inline src/app/code/community/Alma/Installments/Model/System/Config/Backend/PnxStringRender.php:25]:
```suggestion
class Alma_Installments_Model_System_Config_Backend_PnxStringRenderer  extends Mage_Core_Block_Abstract
```

File: src/app/code/community/Alma/Installments/sql/alma_installments_setup/mysql4-install-2.0.0.php [hunk 30]
```diff
@@ -0,0 +1,19 @@
...
+    'payment/alma_installments/p3x_enabled',
+    'payment/alma_installments/p3x_min_amount',
+    'payment/alma_installments/p3x_max_amount',
+    'payment/alma_installments/p4x_enabled',
+    'payment/alma_installments/p4x_min_amount',
+    'payment/alma_installments/p4x_max_amount',
+];
+$table = $installer->getConnection()->getTableName("core_config_data");
+foreach ($pathToDelete as $path) {
+$installer->getConnection()->delete($table,array('path = ?' => $path));
```
Comment (syjust-alma) [inline src/app/code/community/Alma/Installments/sql/alma_installments_setup/mysql4-install-2.0.0.php:17]: are we just remove them or should we do also copy configuration to new `paths` ?

File: src/app/locale/en_US/Alma_Installments.csv [hunk 31]
```diff
@@ -54,4 +54,13 @@
 "Installments count","Installments count"
 "Paid amount (%1) does not match due amount (%2) for order %3","Paid amount (%1) does not match due amount (%2) for order %3"
 "Payment state incorrect (%s & %s) for order %s","Payment state incorrect (%s & %s) for order %s"
-"Order canceled by customer","Order canceled by customer"
\ No newline at end of file
+"Order canceled by customer","Order canceled by customer"
```
Comment (syjust-alma) [inline src/app/locale/en_US/Alma_Installments.csv:57]: did you already updated crowdin with this new source keys ?

**Review by syjust-alma — COMMENTED** [review]
There is an `alma-php-client` copy in the repository ?
in a future release, we should remove it and add it as vendor with composer if possible 😉

File: src/app/code/community/Alma/Installments/Helper/Eligibility.php (hunk 1 above)
Comment (Francois-Gomis) [reply src/app/code/community/Alma/Installments/Helper/Eligibility.php]: done

Comment (Francois-Gomis) [reply src/app/code/community/Alma/Installments/Helper/Eligibility.php]: done

File: src/app/code/community/Alma/Installments/Helper/Eligibility.php (hunk 3 above)
Comment (Francois-Gomis) [reply src/app/code/community/Alma/Installments/Helper/Eligibility.php]: done

File: src/app/code/community/Alma/Installments/Helper/Eligibility.php (hunk 2 above)
Comment (Francois-Gomis) [reply src/app/code/community/Alma/Installments/Helper/Eligibility.php]: merge with  `getEligibleFeePlans()`
Saved after filterEligibleFeePlans

File: src/app/code/community/Alma/Installments/Helper/Eligibility.php (hunk 4 above)
Comment (Francois-Gomis) [reply src/app/code/community/Alma/Installments/Helper/Eligibility.php]: done `loadEligibleFeePlansFromApi`

File: src/app/code/community/Alma/Installments/Helper/FeePlansHelper.php (hunk 5 above)
Comment (Francois-Gomis) [reply src/app/code/community/Alma/Installments/Helper/FeePlansHelper.php]: * @param array $almaFeePlans
     * @return array

File: src/app/code/community/Alma/Installments/Helper/FeePlansHelper.php (hunk 7 above)
Comment (Francois-Gomis) [reply src/app/code/community/Alma/Installments/Helper/FeePlansHelper.php]: done

File: src/app/code/community/Alma/Installments/Helper/FeePlansHelper.php (hunk 9 above)
Comment (Francois-Gomis) [reply src/app/code/community/Alma/Installments/Helper/FeePlansHelper.php]: done

File: src/app/code/community/Alma/Installments/Helper/FeePlansHelper.php (hunk 10 above)
Comment (Francois-Gomis) [reply src/app/code/community/Alma/Installments/Helper/FeePlansHelper.php]: done

File: src/app/code/community/Alma/Installments/Helper/FeePlansHelper.php (hunk 11 above)
Comment (Francois-Gomis) [reply src/app/code/community/Alma/Installments/Helper/FeePlansHelper.php]: done

File: src/app/code/community/Alma/Installments/Helper/Eligibility.php (hunk 13 above)
Comment (Francois-Gomis) [reply src/app/code/community/Alma/Installments/Helper/Eligibility.php:179]: done

File: src/app/code/community/Alma/Installments/Helper/Eligibility.php (hunk 14 above)
Comment (Francois-Gomis) [reply src/app/code/community/Alma/Installments/Helper/Eligibility.php]: done

File: src/app/code/community/Alma/Installments/Helper/Eligibility.php (hunk 15 above)
Comment (Francois-Gomis) [reply src/app/code/community/Alma/Installments/Helper/Eligibility.php]: done and remove :           
/** @var Alma\API\Endpoints\Results\Eligibility $feePlansEligibility */

File: src/app/code/community/Alma/Installments/Block/PaymentForm.php (hunk 16 above)
Comment (Francois-Gomis) [reply src/app/code/community/Alma/Installments/Block/PaymentForm.php]: done

File: src/app/code/community/Alma/Installments/Block/PaymentForm.php (hunk 17 above)
Comment (Francois-Gomis) [reply src/app/code/community/Alma/Installments/Block/PaymentForm.php]: done

File: src/app/code/community/Alma/Installments/Block/PaymentForm.php (hunk 18 above)
Comment (Francois-Gomis) [reply src/app/code/community/Alma/Installments/Block/PaymentForm.php]: done

File: src/app/code/community/Alma/Installments/Block/PaymentForm.php (hunk 19 above)
Comment (Francois-Gomis) [reply src/app/code/community/Alma/Installments/Block/PaymentForm.php]: done

File: src/app/code/community/Alma/Installments/Block/PaymentForm.php (hunk 20 above)
Comment (Francois-Gomis) [reply src/app/code/community/Alma/Installments/Block/PaymentForm.php]: yes it's a timestamp

File: src/app/code/community/Alma/Installments/Helper/Eligibility.php (hunk 21 above)
Comment (Francois-Gomis) [reply src/app/code/community/Alma/Installments/Helper/Eligibility.php:69]: done

File: src/app/code/community/Alma/Installments/Helper/FeePlansHelper.php (hunk 22 above)
Comment (Francois-Gomis) [reply src/app/code/community/Alma/Installments/Helper/FeePlansHelper.php:304]: done

File: src/app/code/community/Alma/Installments/Model/System/Config/Backend/PnxConfig.php (hunk 23 above)
Comment (Francois-Gomis) [reply src/app/code/community/Alma/Installments/Model/System/Config/Backend/PnxConfig.php]: it's protected function

File: src/app/code/community/Alma/Installments/Model/System/Config/Backend/PnxConfig.php (hunk 24 above)
Comment (Francois-Gomis) [reply src/app/code/community/Alma/Installments/Model/System/Config/Backend/PnxConfig.php]: ....

File: src/app/code/community/Alma/Installments/Model/System/Config/Backend/PnxStringRender.php (hunk 29 above)
Comment (Francois-Gomis) [reply src/app/code/community/Alma/Installments/Model/System/Config/Backend/PnxStringRender.php:25]: done

File: src/app/code/community/Alma/Installments/sql/alma_installments_setup/mysql4-install-2.0.0.php (hunk 30 above)
Comment (Francois-Gomis) [reply src/app/code/community/Alma/Installments/sql/alma_installments_setup/mysql4-install-2.0.0.php:17]: For the moment : just remove

File: src/app/locale/en_US/Alma_Installments.csv (hunk 31 above)
Comment (Francois-Gomis) [reply src/app/locale/en_US/Alma_Installments.csv:57]: No i am waiting for Romain QA

**Pull request merged**